xbmc/video/ContextMenus.cpp

```diff
@@ -430,12 +448,26 @@ bool CVideoPlayUsing::Execute(const std::shared_ptr<CFileItem>& itemIn) const
 
 bool CVideoPlayVersionUsing::IsVisible(const CFileItem& item) const
 {
-  return item.HasVideoVersions() || item.HasVideoExtras();
+  return item.HasVideoVersions() && !item.IsVideoAssetNav();
 }
 
 bool CVideoPlayVersionUsing::Execute(const std::shared_ptr<CFileItem>& itemIn) const
 {
   const auto item{std::make_shared<CFileItem>(itemIn->GetItemToPlay())};
+  item->SetProperty("video_asset_type", "version");
+  SetPathAndPlay(item, PlayMode::PLAY_VERSION_USING);
+  return true;
+}
+
+bool CVideoPlayExtraUsing::IsVisible(const CFileItem& item) const
+{
+  return item.HasVideoExtras();
+}
+
+bool CVideoPlayExtraUsing::Execute(const std::shared_ptr<CFileItem>& itemIn) const
+{
+  const auto item{std::make_shared<CFileItem>(itemIn->GetItemToPlay())};
+  item->SetProperty("video_asset_type", "extra");
   SetPathAndPlay(item, PlayMode::PLAY_VERSION_USING);
```

Comment:
Maybe the play mode needs to be renamed now that it is also used for extras?

Reply:
sure